bndtools.core/src/bndtools/central/QueryWhiteboard.java

```diff
@@ -0,0 +1,54 @@
+package bndtools.central;
+
+import org.eclipse.search.ui.IQueryListener;
+import org.eclipse.search.ui.NewSearchUI;
+import org.eclipse.ui.IWorkbench;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.util.tracker.ServiceTracker;
+import org.osgi.util.tracker.ServiceTrackerCustomizer;
+
+/**
+ * This component adds whiteboard facility to IQueryListeners.
+ */
+@Component(immediate = true)
+public class QueryWhiteboard implements ServiceTrackerCustomizer<IQueryListener, IQueryListener> {
+	@Reference
+	IWorkbench										wb;
+
+	ServiceTracker<IQueryListener, IQueryListener>	listeners;
+
+	BundleContext									bc;
+
+	@Activate
+	void activate(BundleContext bc) {
```

Comment:
```suggestion
        @Activate
	public QueryWhiteboard(BundleContext bc) {
```
will cause activation to always occur before any bind methods are fired.